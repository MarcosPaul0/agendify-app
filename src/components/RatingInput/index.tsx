import { View, TouchableOpacity } from 'react-native';
import { Star } from 'phosphor-react-native';
import { COLORS } from '@constants/colors.constant';
import { IRatingInputProps } from './interfaces/ratingInputProps.interface';

export function RatingInput({ onChange, value }: IRatingInputProps) {
  return (
    <View className="flex-row py-1 ">
      {Array.from({
        length: 5,
      }).map((_, index) => {
        const starIsFilled = index < value ? 'fill' : 'regular';
        const key = `rating${index}`;

        return (
          <TouchableOpacity
            key={key}
            testID={key}
            onPress={() => onChange(index + 1)}
          >
            <Star size={24} weight={starIsFilled} color={COLORS.WARNING} />
          </TouchableOpacity>
        );
      })}
    </View>
  );
}
