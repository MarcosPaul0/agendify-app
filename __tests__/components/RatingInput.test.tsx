import { render } from '@testing-library/react-native';
import { RatingInput } from '@components/RatingInput';

describe('Rating Input', () => {
  it('Match with snapshot', async () => {
    const tree = render(<RatingInput onChange={() => null} value={5} />);

    expect(tree).toMatchSnapshot();
  });
});
