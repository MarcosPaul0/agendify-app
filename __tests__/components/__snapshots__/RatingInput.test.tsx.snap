// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating Input Match with snapshot 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "paddingBottom": 4,
        "paddingTop": 4,
      },
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="rating0"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      color="#FFD015"
      fill="#FFD015"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      size={24}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          [
            undefined,
            undefined,
            {},
          ],
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      tintColor="#FFD015"
      vbHeight={256}
      vbWidth={256}
      width={24}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4294955029,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGRect
          fill={null}
          height="256"
          propList={
            [
              "fill",
            ]
          }
          width="256"
          x="0"
          y="0"
        />
        <RNSVGPath
          d="M239.2,97.4A16.4,16.4,0,0,0,224.6,86l-59.4-4.1-22-55.5A16.4,16.4,0,0,0,128,16h0a16.4,16.4,0,0,0-15.2,10.4L90.4,82.2,31.4,86A16.5,16.5,0,0,0,16.8,97.4,16.8,16.8,0,0,0,22,115.5l45.4,38.4L53.9,207a18.5,18.5,0,0,0,7,19.6,18,18,0,0,0,20.1.6l46.9-29.7h.2l50.5,31.9a16.1,16.1,0,0,0,8.7,2.6,16.5,16.5,0,0,0,15.8-20.8l-14.3-58.1L234,115.5A16.8,16.8,0,0,0,239.2,97.4Z"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="rating1"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      color="#FFD015"
      fill="#FFD015"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      size={24}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          [
            undefined,
            undefined,
            {},
          ],
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      tintColor="#FFD015"
      vbHeight={256}
      vbWidth={256}
      width={24}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4294955029,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGRect
          fill={null}
          height="256"
          propList={
            [
              "fill",
            ]
          }
          width="256"
          x="0"
          y="0"
        />
        <RNSVGPath
          d="M239.2,97.4A16.4,16.4,0,0,0,224.6,86l-59.4-4.1-22-55.5A16.4,16.4,0,0,0,128,16h0a16.4,16.4,0,0,0-15.2,10.4L90.4,82.2,31.4,86A16.5,16.5,0,0,0,16.8,97.4,16.8,16.8,0,0,0,22,115.5l45.4,38.4L53.9,207a18.5,18.5,0,0,0,7,19.6,18,18,0,0,0,20.1.6l46.9-29.7h.2l50.5,31.9a16.1,16.1,0,0,0,8.7,2.6,16.5,16.5,0,0,0,15.8-20.8l-14.3-58.1L234,115.5A16.8,16.8,0,0,0,239.2,97.4Z"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="rating2"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      color="#FFD015"
      fill="#FFD015"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      size={24}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          [
            undefined,
            undefined,
            {},
          ],
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      tintColor="#FFD015"
      vbHeight={256}
      vbWidth={256}
      width={24}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4294955029,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGRect
          fill={null}
          height="256"
          propList={
            [
              "fill",
            ]
          }
          width="256"
          x="0"
          y="0"
        />
        <RNSVGPath
          d="M239.2,97.4A16.4,16.4,0,0,0,224.6,86l-59.4-4.1-22-55.5A16.4,16.4,0,0,0,128,16h0a16.4,16.4,0,0,0-15.2,10.4L90.4,82.2,31.4,86A16.5,16.5,0,0,0,16.8,97.4,16.8,16.8,0,0,0,22,115.5l45.4,38.4L53.9,207a18.5,18.5,0,0,0,7,19.6,18,18,0,0,0,20.1.6l46.9-29.7h.2l50.5,31.9a16.1,16.1,0,0,0,8.7,2.6,16.5,16.5,0,0,0,15.8-20.8l-14.3-58.1L234,115.5A16.8,16.8,0,0,0,239.2,97.4Z"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="rating3"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      color="#FFD015"
      fill="#FFD015"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      size={24}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          [
            undefined,
            undefined,
            {},
          ],
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      tintColor="#FFD015"
      vbHeight={256}
      vbWidth={256}
      width={24}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4294955029,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGRect
          fill={null}
          height="256"
          propList={
            [
              "fill",
            ]
          }
          width="256"
          x="0"
          y="0"
        />
        <RNSVGPath
          d="M239.2,97.4A16.4,16.4,0,0,0,224.6,86l-59.4-4.1-22-55.5A16.4,16.4,0,0,0,128,16h0a16.4,16.4,0,0,0-15.2,10.4L90.4,82.2,31.4,86A16.5,16.5,0,0,0,16.8,97.4,16.8,16.8,0,0,0,22,115.5l45.4,38.4L53.9,207a18.5,18.5,0,0,0,7,19.6,18,18,0,0,0,20.1.6l46.9-29.7h.2l50.5,31.9a16.1,16.1,0,0,0,8.7,2.6,16.5,16.5,0,0,0,15.8-20.8l-14.3-58.1L234,115.5A16.8,16.8,0,0,0,239.2,97.4Z"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="rating4"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      color="#FFD015"
      fill="#FFD015"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      size={24}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          [
            undefined,
            undefined,
            {},
          ],
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      tintColor="#FFD015"
      vbHeight={256}
      vbWidth={256}
      width={24}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4294955029,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGRect
          fill={null}
          height="256"
          propList={
            [
              "fill",
            ]
          }
          width="256"
          x="0"
          y="0"
        />
        <RNSVGPath
          d="M239.2,97.4A16.4,16.4,0,0,0,224.6,86l-59.4-4.1-22-55.5A16.4,16.4,0,0,0,128,16h0a16.4,16.4,0,0,0-15.2,10.4L90.4,82.2,31.4,86A16.5,16.5,0,0,0,16.8,97.4,16.8,16.8,0,0,0,22,115.5l45.4,38.4L53.9,207a18.5,18.5,0,0,0,7,19.6,18,18,0,0,0,20.1.6l46.9-29.7h.2l50.5,31.9a16.1,16.1,0,0,0,8.7,2.6,16.5,16.5,0,0,0,15.8-20.8l-14.3-58.1L234,115.5A16.8,16.8,0,0,0,239.2,97.4Z"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
